epochStart/metachain/stakingDataProvider.go

```diff
@@ -18,25 +18,31 @@ import (
 )
 
 type ownerStats struct {
-	numEligible        int
-	numStakedNodes     int64
-	topUpValue         *big.Int
-	totalStaked        *big.Int
-	eligibleBaseStake  *big.Int
-	eligibleTopUpStake *big.Int
-	topUpPerNode       *big.Int
-	blsKeys            [][]byte
+	numEligible          int
+	numStakedNodes       int64
+	numActiveNodes       int64
+	numAuctionNodes      int64
```

Comment:
Also here I think you can remove this variable 
This adds extra complexity because you have to maintain this

Reply:
Right, good suggestion; removed it from here and from stakingDataProvider.
However, I've kept it in auctionListSelector, since there we do a lot of iterations and I don't want to re-calculate len every time.